### ISSUE TITLE
the connection hangs after logging

### ISSUE THREAD

**Issue by DylanBowden** [sazze/winston-logstash-udp#10]
Hello, I have set-up a connection to logstash and the messages are correctly passed. However, the connection doesn't ever close. Debugging the code, I found  a way around the issue but I don't like it even if it works. II created a corresponding pull request

In essence I replaced line 101

```
self.client.send(buf, 0, buf.length, self.port, self.host, callback)
```

by

```
try {
        self.client.send(buf, 0, buf.length, self.port, self.host, function(err) {
          self.client.close();
        });
    } catch (Error) {
    }
```

**Comment by cthayer** [comment]
@HomelessGruesomeMoose I'm not sure we should close the connection after every message that is sent.  If more than one message is sent during the calling code's lifetime, the speed at which logs are sent would be negatively impacted by doing so.

I'm not 100% sure off the top of my head, but isn't winston supposed to handle the shutting down of the client?  Perhaps there's some better way of signaling that the process is done sending logs so we can disconnect?  Still, I'm inclined to leave connection handling up to winston.

**Comment by devmetal** [comment]
Hi, my pr [#15](https://github.com/sazze/winston-logstash-udp/pull/15) will solve this problem in >=0.8 versions. Because the unref method added to socket in 0.9.

**Comment by cthayer** [comment]
I agree.  When pr #15 is merged this issue will be resolved for node > 0.8
